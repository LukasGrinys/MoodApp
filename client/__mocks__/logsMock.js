export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}