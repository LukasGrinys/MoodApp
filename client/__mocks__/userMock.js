export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}