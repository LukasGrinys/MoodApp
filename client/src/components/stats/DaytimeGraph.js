import React from 'react';
import styles from './stats.module.css';

import morningIcon from './morning.svg';
import afternoonIcon from './afternoon.svg';
import eveningIcon from './evening.svg';

import { useTheme } from './../../hoc/ThemeContext';

const DaytimeGraph = ({obj, nightmode}) => {
    const darkTheme = useTheme();

    const returnIconClass = (nm) => {
        if (nm === true) {
            return `${styles.daytime_icon_nightmode}`
        } else {
            return `${styles.daytime_icon}`
        }
    }

    const returnBullet = (rating, left) => {
        if (rating !== "N/A") {
            return (
                <div className={styles.daytime_bullet} style={{bottom:`${rating * 30 -10}px`, left:`${left}`}}>
                    <div className={styles.daytime_bullet_text}>{rating}</div>
                </div>
            )
        } else {
            return null
        }
    }
    return (
        <div className={styles.daytime_canvas}>
            <svg width={"100%"} height={"100%"} overflow={"hidden"}>
            <polyline points={"1,60 300,60"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
            <polyline points={"1,120 300,120"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
            <polyline points={"1,180 300,180"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
            <polyline points={"1,240 300,240"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
            <polyline points={"1,300 300,300"} style={{fill:"none", stroke:"#cccccc", strokeWidth:"1"}}/>
            </svg>
            {returnBullet(obj.morningAverage, null)}
            {returnBullet(obj.afternoonAverage, "48%")}
            {returnBullet(obj.eveningAverage, "82%")}
            <div className={styles.y_axis_number_daytime} style={{top:"-12px"}}>10</div>
            <div className={styles.y_axis_number_daytime} style={{top:"48px", left: "-15px"}}>8</div>
            <div className={styles.y_axis_number_daytime} style={{top:"108px", left:"-15px"}}>6</div>
            <div className={styles.y_axis_number_daytime} style={{top:"168px", left:"-15px"}}>4</div>
            <div className={styles.y_axis_number_daytime} style={{top:"228px", left:"-15px"}}>2</div>
            <div className={styles.daytime_bottom_line}>
            <div className={styles.daytime_time}>7:00</div>
            <div className={styles.daytime_time} style={{left:"33%"}}>13:00</div>
            <div className={styles.daytime_time} style={{left:"66%"}}>20:00</div>
            <div className={styles.daytime_time} style={{left:"100%"}}>00:00</div>
            <img src={morningIcon} className={returnIconClass(darkTheme)} style={{left:"13%", padding: "2px"}} alt="none"/>
            <img src={afternoonIcon} className={returnIconClass(darkTheme)} style={{left:"48%"}} alt="none"/>
            <img src={eveningIcon} className={returnIconClass(darkTheme)} style={{left:"82%"}} alt="none"/>
            </div>    
        </div>
    );
};

export default DaytimeGraph;