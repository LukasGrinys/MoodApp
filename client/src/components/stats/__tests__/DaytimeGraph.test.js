import React from 'react';
import DaytimeGraph from '../DaytimeGraph';
import { shallow } from 'enzyme';
import { testLogs } from '../../../../__mocks__/logsMock';

const wrapper = shallow(<DaytimeGraph logs={testLogs}/>);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});