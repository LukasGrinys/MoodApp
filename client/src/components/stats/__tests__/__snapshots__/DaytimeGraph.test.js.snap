// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="daytime_container"
>
  <p>
    <strong>
      Your mood throughout the day:
    </strong>
  </p>
  <div
    className="daytime_canvas"
  >
    <svg
      height="100%"
      overflow="hidden"
      width="100%"
    >
      <polyline
        points="1,60 300,60"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,120 300,120"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,180 300,180"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,240 300,240"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
      <polyline
        points="1,300 300,300"
        style={
          Object {
            "fill": "none",
            "stroke": "#cccccc",
            "strokeWidth": "1",
          }
        }
      />
    </svg>
    <div
      className="y_axis_number_daytime"
      style={
        Object {
          "top": "-12px",
        }
      }
    >
      10
    </div>
    <div
      className="y_axis_number_daytime"
      style={
        Object {
          "left": "-15px",
          "top": "48px",
        }
      }
    >
      8
    </div>
    <div
      className="y_axis_number_daytime"
      style={
        Object {
          "left": "-15px",
          "top": "108px",
        }
      }
    >
      6
    </div>
    <div
      className="y_axis_number_daytime"
      style={
        Object {
          "left": "-15px",
          "top": "168px",
        }
      }
    >
      4
    </div>
    <div
      className="y_axis_number_daytime"
      style={
        Object {
          "left": "-15px",
          "top": "228px",
        }
      }
    >
      2
    </div>
    <div
      className="daytime_bottom_line"
    >
      <div
        className="daytime_time"
      >
        7:00
      </div>
      <div
        className="daytime_time"
        style={
          Object {
            "left": "33%",
          }
        }
      >
        13:00
      </div>
      <div
        className="daytime_time"
        style={
          Object {
            "left": "66%",
          }
        }
      >
        20:00
      </div>
      <div
        className="daytime_time"
        style={
          Object {
            "left": "100%",
          }
        }
      >
        00:00
      </div>
      <img
        alt="Morning"
        className=""
        src={Object {}}
        style={
          Object {
            "left": "13%",
            "padding": "2px",
          }
        }
      />
      <img
        alt="Afternoon"
        className=""
        src={Object {}}
        style={
          Object {
            "left": "48%",
          }
        }
      />
      <img
        alt="Evenining"
        className=""
        src={Object {}}
        style={
          Object {
            "left": "82%",
          }
        }
      />
    </div>
  </div>
</div>
`;
